### ISSUE TITLE
Feature request : support folders from other package managers or custom hook to manage this by ourself or promote an env var

### ISSUE THREAD

**Issue by stevebeauge** [jasongin/nvs#221]
I use nvs on windows for a few month, and I am very happy, except for one point.

Specifically, I have a bunch of issues with pnpm. The nature of pnpm is to symlink packages from a central store within the user profile.
This folder isn't under the folder managed by nvs and that leads to weird behavior (packages are linked across versions of node).

The pnpm store is located under `~/.pnpm-store` (https://pnpm.io/npmrc#store-dir)

It would be could to have a simple way to differentiate this store for each node version.

I imagine a few approaches :
* explicitly support npm, yarn and pnpm in nvs, which are the most used package managers.
* export the current node version in an env var (or the node home) in order to setup pnpm store with this var
* add the ability to call an arbitrary command each time another node version is used. The command may be a custom script that will change the folders links.

I'm a beginner in the node world, I hope it make sense :)

**Comment by michaelschufi** [comment]
Yes, I would love this as well. This way I can use the best tool for the job. pnpm to manage dependencies and nvs to manage the node versions. 

I think the second approach ("export the current node version in an env var") should suffice. Since [pnpm's store-dir](https://pnpm.io/npmrc#store-dir) setting seems to support environment variables in its path.

**Comment by stevebeauge** [comment]
> Yes, I would love this as well. This way I can use the best tool for the job. pnpm to manage dependencies and nvs to manage the node versions. 
> 
> I think the second approach ("export the current node version in an env var") should suffice. Since [pnpm's store-dir](https://pnpm.io/npmrc#store-dir) setting seems to support environment variables in its path.

Pnpm natively supports [settings node version from config file](https://pnpm.io/npmrc#use-node-version). Since pnpm is my main package manager, nvs became quite useless for me.
I still use it for the few times I have to work with npm

**Comment by michaelschufi** [comment]
Isn't this not just for `pnpm run` and `pnpm node`?

I've also looked at [pnpm env](https://pnpm.io/cli/env). The problem I have with pnpm's approach there, is that the node version switching is global and not per shell session. Which means I cannot have 2 projects with different node versions open and use e.g. version 14 in Project A and version 16 in Project B.

--- 

Buuut, I've found a solution for the problem. Yesterday, I tried to create an env variable which updates when nvs changes the version (the second approach) - and it works very well so far:

1. Update [line `#215` in `lib/use.js`](https://github.com/jasongin/nvs/blob/master/lib/use.js#L215)
    ```js
    // old
    require('./postScript').generate({ 'PATH': envPath, 'NPM_CONFIG_PREFIX': null });

    // new
    require('./postScript').generate({ 'PATH': envPath, 'NPM_CONFIG_PREFIX': null, 'NVS_CURRENT_VERSION': version ? version.toString() : linkName });
    ```
2. Add the `store-dir` setting to `~/.npmrc`
    `store-dir=/home/yourusername/.nvs/${NVS_CURRENT_VERSION}/.pnpm-store`

pnpm now uses the store that corresponds to the current node version .

I'll test for a few days/weeks and will create a PR with the solution.
